src/services/authentication/file/file-based-authentication.ts

```diff
@@ -102,7 +102,7 @@ export class FileBasedAuthentication extends DeepstreamPlugin implements Deepstr
         validateHashingAlgorithm(settings.hash)
       }
     } catch (e) {
-      this.services.logger.fatal(EVENT.PLUGIN_INITIALIZATION_ERROR, 'Validating settings failed for file auth', e.message)
+      this.services.logger.fatal(EVENT.PLUGIN_INITIALIZATION_ERROR, 'Validating settings failed for file auth', { error: e })
```

Comment:
The type error I got here, was that `e.message`, wasn't a valid `MetaData` that the function expected. So the code probably had some (inconsequential?) bug here.

However, the `MetaData` interface allows non-declared keys with `any` value, so I chose to put `e` as the value of a new key `error` here. I hope this is in the spirit of the code!

Reply:
Ok with extending metadata.
Can you double check if we are actually logging metadata on fatal errors? A quick look at the code makes me think we are not. at least in the standard logger

Reply:
Thank you for taking the time to review and comment. I will look into this! 👍

Reply:
If you mean if we add metadata to the individual fatal log statements, it seems that only happens in this instance in `file-based-authentication.ts` above:

![image](https://github.com/deepstreamIO/deepstream.io/assets/67230/3e4f25ff-d93b-482c-a97b-4bf0d00bf38a)

For how any included metadata is handled, I found this, which indicates that metadata apparently is included and sent along in some default logging code: https://github.com/deepstreamIO/deepstream.io/blob/50b2b914ff97b7c913114dc1a611422922dafa42/src/services/logger/pino/pino-logger.ts#L83-L94

Does this answer your question, or were you thinking of something different?

Reply:
thats what I meant, the standard logger does not log metadata on fatal error. created issue #1136 as reminder